base/array.jl

```diff
@@ -656,6 +656,20 @@ function collect_to_with_first!(dest, v1, itr, st)
     return grow_to!(dest, itr, st)
 end
 
+function setindex_widen!(dest::Array{T}, el, i) where T
+    @_inline_meta
+    if el isa T || typeof(el) === T
```

Comment:
The second implies the first, so is unnecessary

Reply:
I would have thought that too but it was in the original code so I thought I must have been missing something?

Reply:
I removed it, we'll see what happens?

Reply:
Perhaps some inference reason. Would be good to find where it got introduced and see if there is a reason given.

Reply:
There's some discussion of it here https://github.com/JuliaLang/julia/pull/25828#discussion_r165004950

Reply:
Could this have caused the time-out on osx?

Reply:
Maybe I should just leave it like it is and open a separate issue...

Reply:
Ok, new issue at https://github.com/JuliaLang/julia/issues/30125